WooCommerce/WooCommerceTests/Tools/InfiniteScroll/PaginationTrackerTests.swift

```diff
@@ -0,0 +1,120 @@
+import XCTest
+@testable import WooCommerce
+
+final class PaginationTrackerTests: XCTestCase {
+    private let paginationTracker = PaginationTracker(pageFirstIndex: 0, pageSize: 30)
+    private let mockDelegate = MockPaginationTrackerDelegate()
+
+    override func setUp() {
+        super.setUp()
+        paginationTracker.delegate = mockDelegate
+    }
+
+    func test_ensuring_next_page_after_syncing_first_page_triggers_delegate_to_sync_two_pages() {
+        // Arrange
+        var syncedPages = [PageInformation]()
+        mockDelegate.onSync = { pageNumber, pageSize, reason, onCompletion in
+            syncedPages.append(PageInformation(pageNumber: pageNumber, pageSize: pageSize, reason: reason))
+            onCompletion?(.success(true))
+        }
+
+        // Action
+        paginationTracker.syncFirstPage()
+        paginationTracker.ensureNextPageIsSynced()
+
+        // Assert
+        XCTAssertEqual(syncedPages, [
+            PageInformation(pageNumber: 0, pageSize: 30, reason: nil),
+            PageInformation(pageNumber: 1, pageSize: 30, reason: nil)
+        ])
+    }
+
+    func test_ensuring_next_page_after_syncing_first_page_without_next_page_is_noop() {
```

Comment:
```suggestion
    func test_ensuring_next_page_after_syncing_first_page_without_next_page_is_no_op() {
```
Maybe separating `noop` in `no_op` it's better understandable?

Reply:
yup that's a good point! updated in https://github.com/woocommerce/woocommerce-ios/pull/2750/commits/83045affdd260821e25cadfc58093307ee21e378